### PR TITLE
feat(cli): validate `--output` flag and refactor

### PR DESCRIPTION
**Pull request by MasonM**
### Motivation

This is a follow-up to https://github.com/argoproj/argo-workflows/pull/13656#pullrequestreview-2341800427. Many of the CLI commands take an `--output` flag, but this wasn't being validated, and there was a lot of duplication. The duplication has led to docs getting out-of-date, e.g. `argo cluster-template get --help` says `Output format. Must be one of: json|yaml|wide`, but it accepts `name` too.

### Modifications
This introduces a simple `EnumFlagValue` type to represent enum flags like `--output` and changes all commands with an `--output` flag to use it. This will cause Cobra to give a helpful error if you pass an invalid value, e.g.:
```
$ ./dist/argo list -o INVALID
Error: invalid argument "INVALID" for "-o, --output" flag: One of: name|json|yaml|wide
```

### Verification

Ran E2E tests and manually ran a few commands with `./dist/argo`

### CODE REVIEW

**Comment by isubasinghe** [comment]
Ah @MasonM I think merging the previous PR introduced a bunch of merge conflicts, sorry about this but could you please address these? Thanks

**Comment by MasonM** [comment]
@isubasinghe Sure, done.

**Comment by MasonM** [comment]
Looks like `E2E Tests (test-cli, mysql, true)` failed because it failed to download k3s:
```
2024-10-04T01:11:55.0066282Z [INFO]  Downloading hash https://github.com/k3s-io/k3s/releases/download/v1.31.0+k3s1/sha256sum-amd64.txt
2024-10-04T01:14:10.5622240Z [ERROR]  Download failed
```

File: docs/cli/argo_archive_list.md [hunk 1]
```diff
@@ -28,7 +28,7 @@ argo archive list [flags]
 ```
       --chunk-size int    Return large lists in chunks rather than all at once. Pass 0 to disable.
   -h, --help              help for list
-  -o, --output string     Output format. One of: json|yaml|wide (default "wide")
+  -o, --output string     Output format. One of: name|json|yaml|wide (default "wide")
```
Comment (isubasinghe) [inline docs/cli/argo_archive_list.md:31]: I was a bit confused initially but this makes sense, allows for code reuse.

**Review by isubasinghe — APPROVED** [review]

**Pull request merged**